activesupport/lib/active_support/cache.rb

```diff
@@ -714,15 +760,17 @@ def write_entry(key, entry, **options)
 
         def serialize_entry(entry, **options)
           options = merged_options(options)
-          if @coder_supports_compression && options[:compress]
-            @coder.dump_compressed(entry, options[:compress_threshold] || DEFAULT_COMPRESS_LIMIT)
+          compress_threshold = options[:compress] ? options[:compress_threshold] : Float::INFINITY
+
+          if @coder_supports_compression
+            @coder.dump_compressed(entry, compress_threshold)
           else
-            @coder.dump(entry)
+            try_compress(@coder.dump(entry), compress_threshold)
           end
         end
 
         def deserialize_entry(payload)
-          payload.nil? ? nil : @coder.load(payload)
+          @coder.load(try_decompress(payload)) unless payload.nil?
```

Comment:
i think it would be preferable if we only compressed the "body" of the cache entry, and not the "header" part, so that we can immediately discard stale entries without having to decompress them. 

It would even allow to instantiate the `Entry` instance with the compressed payload, and lazily uncompress it when accessed.